### ISSUE TITLE
Debugger Support

### ISSUE THREAD

**Issue by hansvonn** [microsoft/vscode-mssql#900]
Is there any integration with the VS Code debugger. It would be pretty simple, e.g. run executes the statements, maybe even a flag to attach the debugger or not. Thank you all for the great implementation.

**Comment by kevcunnane** [comment]
Not at present. This one is a little harder than you might think since SQL Servers debugger protocol is over DCOM (as far as I know) which is Windows specific, and we want the features to work on any platform. It's definitely something on our radar, but since for now it's blocked on dependencies I'll backlog it.

Thank you for the suggestion and we'll keep it open since it's a good request.

**Comment by hansvonn** [comment]
Even executing the run from the Debugger menu might be nice to have, even if step in debugging doesn't work. This would allow the user to quickly select connections using a common interface. I realize you can manage connection profiles, which is great. In any case thank you for the reply and great work.

**Comment by borgdylan** [comment]
@kevcunnane Could it be that the strict dependency on DCOM may change as part of the SQL vNext effort?

**Comment by kevcunnane** [comment]
It won't be in SQL Server 2017. It's worth sending feedback on the SQL Server on Linux channels requesting this. Since it's a larger task it has to be prioritized against other things by the core engine team but in "a" vNext down the line this could be done.

**Comment by AntonC9018** [comment]
I'm trying to debug a C# CLR procedure that I call from a TSQL script. I have naively tried starting the debugger with `Debugger.Launch()`, but that did nothing. Any indications as to how I could attach to the process executing my procedure (the C# part of the code), or at least what topics should I look into to achieve / implement this? I have not programmed anything related to debuggers.

**Comment by FritzHerbers** [comment]
Is there any intention to add SQL debugging to Visual Studio Code in the near future?

**Comment by kburtram** [comment]
The existing SQL debugging stack for SQL Server is currently Windows-only, doesn't support Azure SQL, and requires the use of the sa account (or equivalent sysadmin privileges). Due to these limitations, we haven't brought this debugging technology into VS Code. There have been periodic discussions about updating the SQL engine to support a more modern and cross-platform debugging protocol, but this work isn't currently in progress. I'll keep this issue open to track interest, though the current technical constraints make implementing debugging in VS Code less practical at this time.

**Comment by kburtram** [comment]
Thanks for this suggestion! I'm closing older feature requests that aren't planned for an upcoming iteration.  Please see previous comment for details on the challenges with brining this feature to VS Code using the current debugger tech stae. I've added a label so we can reconsider when doing longer-term planning in this area.
